lib/portlayer/storage/volume_cache.go

```diff
@@ -35,22 +36,68 @@ type VolumeLookupCache struct {
 	vlc     map[string]Volume
 	vlcLock sync.RWMutex
 
-	// The underlying data storage implementation
-	volumeStore VolumeStorer
+	// Maps the service url of the volume store to the underlying data storage implementation
+	volumeStores map[url.URL]VolumeStorer
 }
 
-func NewVolumeLookupCache(op trace.Operation, vs VolumeStorer) (*VolumeLookupCache, error) {
+func NewVolumeLookupCache(op trace.Operation) *VolumeLookupCache {
 	v := &VolumeLookupCache{
-		vlc:         make(map[string]Volume),
-		volumeStore: vs,
+		vlc:          make(map[string]Volume),
+		volumeStores: make(map[url.URL]VolumeStorer),
 	}
 
-	return v, v.rebuildCache(op)
+	return v
 }
 
-// List the configured volume stores
-func (v *VolumeLookupCache) VolumeStoresList(op trace.Operation) (map[string]url.URL, error) {
-	return v.volumeStore.VolumeStoresList(op)
+// AddStore adds a volumestore by name.  The url returned is the service url to the volume store.
+func (v *VolumeLookupCache) AddStore(op trace.Operation, storeName string, vs VolumeStorer) (*url.URL, error) {
```

Comment:
@vburenin I think this should be made a separate ticket if we really want to do this. We have a lot of tech debt discussion surrounding the changes to logging in general. I think that it is better to service that request along those lines rather than in this PR.